src/components/dropdownSearch/DropdownSearchReducers.tsx

```diff
@@ -178,29 +198,51 @@ export const dropdownSearchReducer = (state: IDropdownSearchState = dropdownSear
         setFocusOnDropdownButton: false,
       };
     case DropdownSearchActions.filter:
+      const options = (state.options || []);
+      const shouldReturnNewOptions: boolean = state.supportSingleCustomOption && options
+        .filter((option: IDropdownOption) => !option.custom && !option.default)
+        .every((option: IDropdownOption) => (option.displayValue || option.value).toLowerCase() !== (action.payload.filterText || '').toLowerCase());
+
+      const newCustomOption: IDropdownOption[] = action.payload.filterText !== ''
+        ? [{ value: action.payload.filterText, selected: false, custom: true, hidden: false }]
+        : [];
+
       return {
         ...state,
         id: action.payload.id,
+        options: shouldReturnNewOptions
+          ? [...newCustomOption, ...removeCustomOptions(options, state.supportSingleCustomOption, false)]
+          : deepClone(options),
         filterText: action.payload.filterText,
         activeOption: getFilteredOptions(state, action.payload.filterText)[0] || state.activeOption,
         setFocusOnDropdownButton: false,
       };
     case DropdownSearchActions.select:
-      return {
-        ...state,
-        id: action.payload.id,
-        options: selectSingleOption(state.options, action.payload.addedSelectedOption),
-        isOpened: false,
-        activeOption: undefined,
-        setFocusOnDropdownButton: false,
-      };
+      return !state.supportSingleCustomOption
```

Comment:
Only your options change in your object.  You can create a const options and add your logic on him and return this new options in your object